src/Analysis/Engine/Impl/Interpreter/Ast/AstPythonType.cs

```diff
@@ -232,8 +236,8 @@ ILocationInfo loc
 
         public IEnumerable<string> GetMemberNames(IModuleContext moduleContext) {
             var names = new HashSet<string>();
-            lock (_members) {
-                names.UnionWith(_members.Keys);
+            lock (Members) {
```

Comment:
I really don't like an idea of taking locks on mutable, non-private field. `AstPythonBuiltinType` needs it only inside `IsHidden`, which can be fixed by adding
```cs
protected bool ContainsMember(string name) {
    lock (_members) {
        return Members.ContainsKey(name);
    }
}
```

Reply:
OK